cpp/src/replace/replace.cu

```diff
@@ -861,6 +863,49 @@ std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
 }  // namespace experimental
 }  // namespace cudf
 
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+std::unique_ptr<column> replace_nans(column_view const& input,
+                                     column_view const& replacement,
+                                     cudaStream_t stream,
+                                     rmm::mr::device_memory_resource* mr)
+{
+  auto nan_mask = is_not_nan(input, mr, stream);
+  return copy_if_else(input, replacement, nan_mask->view(), mr, stream);
+}
+
+std::unique_ptr<column> replace_nans(column_view const& input,
+                                     scalar const& replacement,
+                                     cudaStream_t stream,
```

Comment:
Is there a particular reason change in the order of stream and memory manager, just out of curiosity nothing else. Doesn't cause any issue.

Reply:
I noticed that since detail APIs are called within other libcudf APIs, we could use them for temporary allocations like here https://github.com/rapidsai/cudf/blob/c37968fab5599f1be734819381f811f3ae3c0945/cpp/src/replace/replace.cu#L875
This requires that one specifies the use of `default_resource`. But the stream parameter is always used. So it made sense to make memory resource _**more**_ optional than stream.